### PR TITLE
Fix different --bs=1 behavior

### PR DESCRIPTION
**Pull request by renxida**
https://xilinx.slack.com/archives/C079V23F7CP/p1729805729161599

A lot of us newcomers are running into issues where we expect a bs=1 vmfb to be a drop in replacement for a bs=4 vmfb, but it's not.

I think we should do a different cmdline arg for serving local.

### CODE REVIEW

File: sharktank/sharktank/examples/export_paged_llm_v1.py [hunk 1]
```diff
@@ -37,6 +43,20 @@ def main():
         help="Output file path for exported config file",
         default="/tmp/batch_llama_v1.json",
     )
+    parser.add_argument(
+        "--kv_cache_type",
+        help="Type of KV cache to use (direct or paged). Default is paged for performant serving. For local use, you may need to set this to direct.",
+        type=str,
+        choices=["direct", "paged"],
+        default="paged",
```
Comment (dan-garvey) [inline sharktank/sharktank/examples/export_paged_llm_v1.py:51]: duplicate

**Review by dan-garvey — CHANGES_REQUESTED** [review]
Lets move this flag to the cli.py file so it can be shared and not repeated. Also, iirc paged is probably not required in the bs1 case?

File: sharktank/sharktank/examples/export_paged_llm_v1.py
Comment (ScottTodd) [inline sharktank/sharktank/examples/export_paged_llm_v1.py:1]: My general feeling with this script is that it is due for a rewrite. That might just involve splitting into separate scripts, one entry point for each model / variant, or something more like a fully self-contained model import pipeline from HF/GGUF --> MLIR --> VMFB+IRPA.

What users will expect is a simple process to follow for off-the-shelf and custom models that produces files which are ready for serving via shortfin. Whatever we need to do with cache types, batch sizes, etc. to get no errors, good accuracy, and good performance should all be handled for the user on the default path. For power users, we can expose tuning and customization options.

Comment (gpetters-amd) [reply sharktank/sharktank/examples/export_paged_llm_v1.py:1]: > or something more like a fully self-contained model import pipeline from HF/GGUF --> MLIR --> VMFB+IRPA.

Are you envisioning something like turbine-models' export scripts or something more akin to an SD pipeline class? I've refactored some of the export_paged_llm_v1 code into a dirty export script I've been working off of that I can clean up if that's the direction we want to move toward.

**Comment by renxida** [comment]
Superceded by #330 thanks Dan!

File: sharktank/sharktank/examples/export_paged_llm_v1.py
Comment (ScottTodd) [reply sharktank/sharktank/examples/export_paged_llm_v1.py:1]: I'm thinking of something between what we have today and https://docs.vllm.ai/en/latest/getting_started/quickstart.html#offline-batched-inference

image

or https://nvidia.github.io/TensorRT-LLM/quick-start-guide.html
image

**Pull request closed without merging**